phantombuster/web-browser-extension
# CP-220 : update manifest v3

## 📝 What is this PR

This PR updates the version of the manifest to upgrade from V2 to V3. The new version is required by Google Chrome starting next year.

-   [Specs](https://phantombuster.atlassian.net/browse/CP-220)

## 🏭 How does it work

Google has a guide to migrate from V2 to V3 [https://developer.chrome.com/docs/extensions/mv3/mv3-migration/](https://developer.chrome.com/docs/extensions/mv3/mv3-migration/). The manifest and some files of the code has been updated. There's also the change from "webextension-polyfill-ts" to "@types/webextension-polyfill" because the first one is deprecated now.

## 🗒 Notes
This version is not functional on Firefox at the moment. Firefox should support Manifest V3 at the end of the year "normally".

## 🧪 QA Process

To test the extension :

npm i
npm run clean
npm run build:dev
On Google Chrome, open chrome://extensions/
Enable the Developer mode
Click on Load Unpacked
Select the dev-build folder
The extension should be activated in the browser (don't forget to disable the official extension)

### ✅ Positive testing (should succeed)

-  To test the update :

**Test when the user is already connected to the service:**
- Use a phantom
- Click on the "Connect to XXXX" button to get the cookie
- The phantom should work

**Test when the user is not connected to the service:**
- Use a phantom
- Click on the "Connect to XXXX" button. You should have a new tab opened
- Log in
- You are redirected to the PhantomBuster page

**Test to click the "Connect to XXXX" with Shift pushed**
- Use a phantom
- Click on the "Connect to XXXX" button with Shift pushed
- Nothing should change


## 👌 To check

-   [ ] There's no regression
-   [ ] Coding guidelines are followed ([Contributing](https://github.com/phantombuster/engineering-docs/blob/master/contributing.md), [Testing](https://github.com/phantombuster/engineering-docs/blob/master/testing.md))
